miden-tx/src/tests/kernel_tests/test_tx.rs

```diff
@@ -579,3 +598,318 @@ fn test_build_recipient_hash() {
         "recipient hash not correct",
     );
 }
+
+#[test]
+fn test_load_foreign_account_basic() {
+    // GET ITEM
+    // --------------------------------------------------------------------------------------------
+    let storage_slot = AccountStorage::mock_item_0().slot;
+    let (foreign_account, _) = AccountBuilder::new(ChaCha20Rng::from_entropy())
+        .add_storage_slot(storage_slot.clone())
+        .code(AccountCode::mock_account_code(TransactionKernel::testing_assembler(), false))
+        .nonce(ONE)
+        .build()
+        .unwrap();
+
+    let account_id = foreign_account.id();
+    let mock_chain = MockChainBuilder::default().accounts(vec![foreign_account.clone()]).build();
+    let advice_inputs = get_mock_advice_inputs(&foreign_account, &mock_chain);
+
+    let tx_context = TransactionContextBuilder::with_standard_account(ONE)
+        .mock_chain(mock_chain)
+        .advice_inputs(advice_inputs.clone())
+        .build();
+
+    let code = format!(
+        "
+        use.kernel::prologue
+        use.miden::tx
+        use.miden::account
+        use.miden::kernel_proc_offsets
+
+        begin
+            exec.prologue::prepare_transaction
+
+            # pad the stack for the `execute_foreign_procedure`execution
+            padw padw push.0.0.0
+            # => [pad(11)]
+
+            # push the index of desired storage item
+            push.0
```

Comment:
After the implementation of the storage bounds (#886) account procedures can only access the 0'th memory slot, so right now I can test the `get_item_foreign` and `get_map_item_foreign` procedures only with `0` slot index. I think we (I) should rework these tests after the correct value of the storage bound will be assigned to the each account procedure to make sure that we can obtain values in other storage slots.

Reply:
Let's create an issue for this.

Reply:
Created an issue #917.